delta/plugins/search/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/search/models/SearchRejection.scala

```diff
@@ -0,0 +1,48 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.search.models
+
+import akka.http.scaladsl.model.StatusCodes
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.ClassUtils
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.ContextValue
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.encoder.JsonLdEncoder
+import ch.epfl.bluebrain.nexus.delta.sdk.http.HttpClientError
+import ch.epfl.bluebrain.nexus.delta.sdk.marshalling.HttpResponseFields
+import io.circe.syntax._
+import io.circe.{Encoder, JsonObject}
+
+/**
+  * Enumeration of search rejection types.
+  *
+  * @param reason a descriptive message as to why the rejection occurred
+  */
+sealed abstract class SearchRejection(val reason: String) extends Product with Serializable
+
+object SearchRejection {
+
+  /**
+    * Signals a rejection caused when interacting with the elasticserch client
+    */
+  final case class WrappedElasticSearchClientError(error: HttpClientError)
```

Comment:
If we are going to use an underlying aggregate elasticsearch view, we don't need to define another rejection no ?
I suppose we will use `ElasticSearchViewsQuery` to return the results and it already defines an error for that

Reply:
We could try to use the aggregate elasticsearch view. But do we need it? The only thing we need is to know the projects a caller has access to, which can be easily done with `projects.list` with the [appropriate filtering](https://github.com/BlueBrain/nexus/blob/master/delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutes.scala#L60).

Afterwards just run your query:

```scala
client.search(query, projectsToIndices)(...)
```

There is no need of something more complicated right now. You don't need to construct an aggregate elasticsearch view (which anyway they do not currently support compositeview projections).